patches/server/0061-Complete-resource-pack-API.patch

```diff
@@ -45,7 +45,16 @@ index 87a193e8f026166b5821710d5501eb11175744b0..21df67c4b78947868407d96cd91a3ce0
  
      public CraftPlayer(CraftServer server, ServerPlayer entity) {
          super(server, entity);
-@@ -1941,6 +1946,45 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+@@ -1494,7 +1499,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+ 
+     @Override
+     public void setResourcePack(String url) {
+-        this.setResourcePack(url, null);
++        this.setResourcePack(url, (byte[]) null);
```

Comment:
Upstream added the same API as us ([Bukkit](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/0c7075426331703c128edb648193c57fdd98aa4e)/[CB](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/439c685f4ea3452ce07adad6f4da7526daac2eac) commits), what should I do here?

Reply:
Maybe remove the methods added in paper (in teory the basics setResourcePack, the adventure method can still) the issue is because here the method with hash is NotNull and is String.. and in Bukkit is byte[] and can be null.. maybe keep all, can be?